src/sql/azdata.proposed.d.ts

```diff
@@ -387,11 +388,23 @@ declare module 'azdata' {
 		selectedRow?: number;
 	}
 
+
+	export interface DeclarativeTableMenuCellValue {
+		/**
+		 * commands for the menu. Use an array for a group and menu separators will be added.
+		 */
+		commands: (string | string[])[];
+		/**
+		 * context that will be passed to the commands.
+		 */
+		context: any
```

Comment:
Not sure we want to allow passing anything as the context - there's tons of stuff that will error out when it tries to go over the JSON RPC and we should at least try to tell users what stuff is safe to send over. 

We should have a discussion about what this "context" entails, but could we not just have a simple key:value mapping to pass over or something?

Reply:
yeah, let me restrict the supported types here.